policy/lib/image/image.rego

```diff
@@ -7,9 +7,11 @@ import rego.v1
 parse(ref) := d if {
 	digest_parts := split(trim_space(ref), "@")
 
-	contains(digest_parts[0], "/")
 	repo_parts := split(digest_parts[0], "/")
 
+	# match hostname. i.e registry.local
+	regex.match("^(?:[a-zA-Z0-9-])+\\.[a-zA-Z]+", repo_parts[0])
```

Comment:
Unicode characters, although unusual, are valid host names.

Reply:
If you think it's too hard to properly match all valid host names, maybe it's acceptable with a comment, e.g.:
```
  # This incorrectly disallows host names with unicode chars
```

But, (as per my other comment), can you explain why are we're doing this?

Reply:
My guess is that EC is incorrectly assuming a certain string is a hostname when it's something else completely.

This is mentioned in the docs earlier in this file, but just in case it was missed, this check was re-implemented here from OSBS, see their [docs](https://osbs.readthedocs.io/en/latest/users.html#pullspec-locations) which also link to the implementation: https://github.com/containerbuildsystem/atomic-reactor/blob/14b9ad84b281029391f1bb83dc5dee6c573eeb4b/atomic_reactor/utils/operator.py#L316

I believe `alnum` will actually handle unicode characters as well.

Reply:
The Jira didn't describe it well. `image.parse` was parsing `'{"kind":"Namespace","apiVersion":"v1","metadata":{"name":"openshift-workload-availability","annotations":{"openshift.io/node-selector":""}}}'` and returning `"annotations":{"openshift.io/node-selector":""}` as a repo.

Do we care about unicode? I think adding the comment that it disallows it would be enough?

Reply:
Ooooh, so the entire string was being treated like it might be an image ref.

Reply:
yeah. I'm also finding a solid regex for matching hosts is ugly, so I thought I'd start with this.

Reply:
But why was it returning `"annotations":{"openshift.io/node-selector":""}`. Because of the comma?

Reply:
Alternative idea: We could flip it and look for "{" or "}" chars, and if they're found we assume it's not a url.

Reply:
..with some explanatory comments to explain why.

Reply:
(Not saying that's better, just something to consider.)

Reply:
Are there reservations about this change or can it move forward? I added a few more tests if it helps. https://github.com/enterprise-contract/ec-policies/pull/1347/commits/82ab6d5ce93fd6f41d4491723390926249f085c4